src/components/Chat/hooks/useConnectionListener.js

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState } from 'react';
+
+import { NetInfo } from '../../../native';
+
+export const useConnectionListener = ({ client }) => {
+  const [unsubscribeNetInfo, setUnsubscribeNetInfo] = useState(() => null);
```

Comment:
??
```suggestion
  const [unsubscribeNetInfo, setUnsubscribeNetInfo] = useState(null);
```

Reply:
was just thinking ahead to TS here since this will be a function once set, but could either do a union on null in the state hook declaration or on the function type itself once typed

Reply:
gotcha, makes sense. I don't know if `() => null` returns the function type for useState. on the docs for hooks this is the style for setting up a state hook that requires longer running code than a simple set